spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -125,6 +124,39 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
     }.toDF("path")
   }
 
+  def getFilesFromReservoir(basePath: String,
+                            partitionColumns: Seq[String],
+                            reservoir: DataFrame): Dataset[SerializableFileStatus] = {
+    val expectedSchema = StructType(
+      Seq(
+        StructField("path", StringType),
+        StructField("length", LongType),
+        StructField("isDir", BooleanType),
+        StructField("modificationTime", LongType)
+      ))
+
+    if(reservoir.schema != expectedSchema) {
+      throw new SparkException(s"reservoir schema should match ${expectedSchema.treeString}")
+    }
+
+    implicit val fileNameAndSizeEncoder
+    = org.apache.spark.sql.Encoders.product[SerializableFileStatus]
+
+    reservoir
+      .filter(s"path like '$basePath/%' ")
```

Comment:
Can the base path contain a `%` or `_`? That would confuse the `like`. Maybe something like:
```suggestion
      .filter(startsWith(col("path"), lit(s"$basePath/")))
```

Reply:
I am not sure if base paths can have `%`. At least on AWS, this is least likely as it falls under special characters that need extreme special handling and would often create problems in various AWS services. I think your suggestion is cleaner and I shall update the logic to use `startsWith` function.